internal/api/rbac/roles.go

```diff
@@ -132,6 +133,13 @@ func (r *rolesDatabase) Create(
 		)
 	}
 
+	// Append the description annotation to the Role if it exists
+	if annotations := kargoRole.GetAnnotations(); annotations != nil {
+		if description, ok := annotations[kargoapi.AnnotationKeyDescription]; ok {
+			sa.Annotations[kargoapi.AnnotationKeyDescription] = description
+		}
+	}
```

Comment:
```suggestion
	if description, ok := kargoRole.Annotations[kargoapi.AnnotationKeyDescription]; ok {
		sa.Annotations[kargoapi.AnnotationKeyDescription] = description
	}
```

The `ok` check already covers for the annotations potentially being `nil`.